chaincode/marbles_transfer/marbles_transfer.go

```diff
@@ -348,22 +391,88 @@ func transferMarbleState(ctx contractapi.TransactionContextInterface, marble *Ma
 	}
 
 	// Transfer the private properties (delete from seller collection, create in buyer collection)
-
-	collectionSeller := "_implicit_org_" + clientOrgID
+	collectionSeller := buildCollectionName(clientOrgID)
 	err = ctx.GetStub().DelPrivateData(collectionSeller, marble.ID)
 	if err != nil {
 		return fmt.Errorf("failed to delete Marble private details from seller: %s", err.Error())
 	}
 
-	collectionBuyer := "_implicit_org_" + buyerOrgID
+	collectionBuyer := buildCollectionName(buyerOrgID)
 	err = ctx.GetStub().PutPrivateData(collectionBuyer, marble.ID, immutablePropertiesJSON)
 	if err != nil {
 		return fmt.Errorf("failed to put Marble private properties for buyer: %s", err.Error())
 	}
 
-	// TODO delete the price records for buyer and seller
+	// Delete the price records for seller
+	marblePriceKey, err := ctx.GetStub().CreateCompositeKey(typeMarbleForSale, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key for seller: %s", err.Error())
+	}
 
-	// TODO add a state record for a 'receipt' in both buyer and seller private data collection to record the sales price and date
+	err = ctx.GetStub().DelPrivateData(collectionSeller, marblePriceKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete marble price from implicit private data collection for seller: %s", err.Error())
+	}
+
+	// Delete the price records for buyer
+	marblePriceKey, err = ctx.GetStub().CreateCompositeKey(typeMarbleBid, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key for buyer: %s", err.Error())
+	}
+
+	err = ctx.GetStub().DelPrivateData(collectionBuyer, marblePriceKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete marble price from implicit private data collection for buyer: %s", err.Error())
+	}
+
+	// Keep record for a 'receipt' in both buyer and seller private data collection to record the sales price and date
+	// Persist the agreed to price in a collection sub-namespace based on receipt key prefix
+	receiptBuyKey, err := ctx.GetStub().CreateCompositeKey(typeMarbleBuyReceipt, []string{ctx.GetStub().GetTxID(), marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key for receipt: %s", err.Error())
+	}
+
+	// Create the receipt
+	agreementJSON, err := getAssetPrice(ctx, marble.ID, typeMarbleForSale)
```

Comment:
This will retrieve the price json from the ledger. That seems like an unnecessary ledger hit, since the price json was already passed into TransferAsset(). I'd recommend simply pass it into transferMarbleState() instead of looking up from ledger again.

Reply:
🛠️